docs/src/literate-tutorials/dg_heat_equation.jl

```diff
@@ -0,0 +1,397 @@
+#     versions. However, through the code we use `f` and `u` instead to reflect the strong
+#     connection between the weak form and the Ferrite implementation.
+
+function assemble_global(cellvalues::CellValues, facevalues::FaceValues, interfacevalues::InterfaceValues, K::SparseMatrixCSC, dh::DofHandler, h::Float64)
+    ## Allocate the element stiffness matrix and element force vector
+    n_basefuncs = getnbasefunctions(cellvalues)
+    Ke = zeros(n_basefuncs, n_basefuncs)
+    fe = zeros(n_basefuncs)
+    Ki = zeros(n_basefuncs * 2, n_basefuncs * 2)
+    ## Allocate global force vector f
+    f = zeros(ndofs(dh))
+    ## Create an assembler
+    assembler = start_assemble(K, f)
+    ## Loop over all cells
+    for cell in CellIterator(dh)
+        ## Reinitialize cellvalues for this cell
+        reinit!(cellvalues, cell)
+        ## Compute volume integral contribution
+        assemble_element!(Ke, fe, cellvalues)
+        ## Assemble Ke and fe into K and f
+        assemble!(assembler, celldofs(cell), Ke, fe)
+    end
+    ## get interpolation dimensions and order to use in the penalty weight, using the $(1 + order)^{dim}$ for $\eta$
+    ip = Ferrite.function_interpolation(facevalues)
+    order = Ferrite.getorder(ip)
+    dim = Ferrite.getdim(ip)
+    μ = (1 + order)^dim / h
+    ## Loop over all interfaces
+    for ic in InterfaceIterator(dh)
+        ## Reinitialize interfacevalues for this interface
+        reinit!(interfacevalues, ic)
+        ## Compute interface surface integrals contribution
+        assemble_interface!(Ki, interfacevalues, μ)
+        ## Assemble Ki into K
+        assemble!(assembler, interfacedofs(ic), Ki)
+    end
+    ## Loop over domain boundaries with Neumann boundary conditions
+    for fc in FaceIterator(dh, ∂Ωₙ)
+        ## Reinitialize face_values_a for this boundary face
+        reinit!(facevalues, fc)
+        ## Compute boundary face surface integrals contribution
+        assemble_boundary!(fe, facevalues)
+        ## Assemble fe into f
+        assemble!(f, celldofs(fc), fe)
+    end
+    return K, f
+end
+#md nothing # hide
+
+# ### Solution of the system
+# The last step is to solve the system. First we call `assemble_global`
+# to obtain the global stiffness matrix `K` and force vector `f`.
+K, f = assemble_global(cellvalues, facevalues, interfacevalues, K, dh, h);
+
+# To account for the boundary conditions we use the `apply!` function.
+# This modifies elements in `K` and `f` respectively, such that
+# we can get the correct solution vector `u` by using `\`.
+apply!(K, f, ch)
+u = K \ f;
+
+# ### Exporting to VTK
+# To visualize the result we export the grid and our field `u`
+# to a VTK-file, which can be viewed in e.g. [ParaView](https://www.paraview.org/).
+vtk_grid("dg_heat_equation", dh) do vtk
+    vtk_point_data(vtk, dh, u)
+end
+
+## test the result                #src
+#using Test                        #src
+#@test norm(u) ≈ 3.307743912641305 #src
```

Comment:
It would be good to test: Update the result to what you get so that any changes resulting in different results are detected.